### DISCUSSION TITLE
关于附录插入的问题

### DISCUSSION THREAD

**Discussion by HgHe-xdu** [note286/xduts#65] [Q&A] [answered]
用Matlab输出了代码的PDF文件，请问插入附录是以内嵌EPS图片的形式插入吗？

**Comment by note286** [comment]
请更详细的描述你的需求。

**Reply by HgHe-xdu** [reply]
附录以tex的形式插入，那想要在附录下添加PDF的内容 是以图片的形式插入吗？

**Reply by note286** [reply]
也有其他方式，以图片方式插入是可以的。

**Reply by HgHe-xdu** [reply]
大佬可以介绍一下其他方法吗 图片插入排版有一点问题 谢谢！

**Reply by note286** [reply]
根据你的信息暂时无法提供相应的解决方案，目前无法明晰你的需求细节。

**Reply by HgHe-xdu** [reply]
如何将PDF的内容原原本本地插入附录中，PDF大概四页

**Reply by note286** [reply]
`\includegraphics`是有`page`参数的。

但是我觉得你的本质问题不是如何插入多页PDF，不过你既然这么问了，我就不解答多余的了。

**Comment by note286** [comment]
如果是排版代码，推荐使用`listings`或者`minted`宏包。

**Comment by note286** [comment, marked as answer]
我猜测你想将MATLAB代码排版，我也可能猜错了。

```latex
% appendix.tex
\chapter{MATLAB示例}
\inputminted{matlab}{LDA.m}
```

```latex
% main.tex
\documentclass{xdupgthesis}
\xdusetup{info/appendix=appendix.tex}
\usepackage{minted}
\setminted{linenos=true,numbersep=.5ex,breaklines,breakanywhere}
\begin{document}
\end{document}
```

`LDA.m` 代码来自 [https://github.com/biometrics/openbr/blob/master/scripts/matlab/LDA.m](https://github.com/biometrics/openbr/blob/master/scripts/matlab/LDA.m) 。

```matlab
function [subspaceData]=LDA(X,classNo,varargin)
    useFixedEnergy = false;
    doDominantEig = true;
    doPCA = true;
    useFixedEig = false;
    energyPercentage = .98;
    ScaleW = 1;
    doDLDA = false;
    do_null = false;
    ...
    subspaceData.mean = Mean_Vector(:);
    subspaceData.mean2 = mean2(:);
    subspaceData.W1 = Select_eigenvectors;
    subspaceData.D1 = eigenvalues;
    subspaceData.W2 = SW_vec;
    subspaceData.W3 = SB_vec;
    subspaceData.W = (subspaceData.W3' * subspaceData.W2' * subspaceData.W1')';
```

![image](https://user-images.githubusercontent.com/16207292/224464509-edfc6476-5fb6-4191-a2dd-b08c11087e58.png)

**Reply by HgHe-xdu** [reply]
谢谢大佬 那我采用这个方法编写 谢谢！

**Reply by note286** [reply]
两点：

一、起码给个star啊；

二、描述问题要清晰，不要问XY问题，都已经是研究生毕业的人了，你此次的需求可以简单描述为，我有很长的MATLAB代码，我想要美观地放在在附录中。而不是如何插入多页PDF，甚至如何插入多页PDF这一点都没有描述出来，还是我问出来的。

**Reply by HgHe-xdu** [reply]
好！！！
